drivers/gpio/gpio_silabs.c

```diff
@@ -303,15 +304,19 @@ static void gpio_silabs_common_isr(const struct device *dev)
 	uint32_t int_status = GPIO->IF;
 
 	ARRAY_FOR_EACH(data->ports, i) {
-		if (!int_status) {
-			break;
-		}
-		const struct device *port_dev = data->ports[i];
-		struct gpio_silabs_port_data *port_data = port_dev->data;
-		uint32_t enabled_int = int_status & port_data->int_enabled_mask;
+		const struct device *port_dev;
+		const struct gpio_silabs_port_config *port_config;
+		struct gpio_silabs_port_data *port_data;
+		uint32_t enabled_int;
+		uint32_t pin_mask;
+
+		port_dev = data->ports[i];
+		port_config = port_dev->config;
+		port_data = port_dev->data;
+		enabled_int = int_status & sl_hal_gpio_get_enabled_interrupts();
 
 		if (enabled_int) {
-			uint32_t pin_mask = 0;
+			pin_mask = 0;
 
 			ARRAY_FOR_EACH(data->irq_pin_map, int_line) {
 				if ((enabled_int & BIT(int_line)) &&
```

Comment:
In fact, I think this condition is sufficient. If `data->irq_pin_map[int_line] == 0xFF`, we have a trouble. You can just have an assert.